runsc/container/container.go

```diff
@@ -973,7 +975,7 @@ func (c *Container) initGoferConfs(ovlConf config.Overlay2, mountHints *boot.Pod
 				mountType = hint.Mount.Type
 			}
```

Comment:
If the mount is being configured via mount annotations (when hint != nil), then the overlaySize should be reset. We should have not parts of the configuration coming from the hint and parts coming from the flag. Would be confusing. If we need to set size limits on mount-annotations-configured mounts, then we will add size in the mount annotations in the future.